src/components/DigitalConverterComponent.cpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * @file DigitalConverterComponent.cpp
+ * @author Luciana Moreno Borges (lumoreno.borges@gmail.com)
+ * @brief Digital converter component: receives a sinusoidal value and converts on a digital one
+ * @version 0.1
+ * @date 2019-01-25
+ * 
+ * @copyright Copyright (c) 2019
+ * 
+ */
+#include <hyro/DigitalConverterComponent.h>
+
+#include <hyro/utils/SpinnerDefault.h>
+#include <hyro/utils/SpinnerRated.h>
+#include <hyro/factory/CommandFactory.h>
+
+#include <hyro/msgs/common/Basic.h>
+#include <hyro/Thresholding.h>
+#include <hyro/msgs/Signal.h>
+
+#include <iostream>
+
+namespace hyro
+{
+  DigitalConverterComponent::DigitalConverterComponent(URI uri): hyro::Component(uri){
+
+  }
+
+  std::shared_ptr<hyro::HyroLogger> DigitalConverterComponent::s_logger = hyro::HyroLoggerManager::CreateLogger("DigitalConverterComponent");
+  
+  /*Initialize the input and outputs of the object*/
+  hyro::Result
+  DigitalConverterComponent::init (const hyro::ComponentConfiguration & config)
+  {
+    m_input = this->registerInput<Signal>("signals"_uri, config);
+    m_output = this->registerOutput<float>("digital_signals"_uri, config);
+    s_logger->info("Init:");
+    return hyro::Result::RESULT_OK;
+  }
+
+  /*State transisitions*/
+  hyro::Result
+  DigitalConverterComponent::reset ()
+  {
+    m_input.reset();
+    m_output.reset();
+    return hyro::Result::RESULT_OK;
+  }
+
+  hyro::Result
+  DigitalConverterComponent::check ()
+  {
+    return hyro::Result::RESULT_OK;
+  }
+
+  /*Define a callback method to receive the sinusoidal signal value*/
+  hyro::Result
+  DigitalConverterComponent::start ()
+  {
+    m_spinner = SpinnerDefault::Create(m_input, &DigitalConverterComponent::callback, this, this->suspensionToken());
+    return hyro::Result::RESULT_OK;
+  }
+
+  hyro::Result
+  DigitalConverterComponent::update()
+  {
+    s_logger->info("Converting signal ...");
+    return hyro::Result::RESULT_OK;
+  }
+
+  /*Callback function to receive the sinusoidal signal value , configure its parameters 
+  and send to another component */
+  void
+  DigitalConverterComponent::callback (std::shared_ptr<const Signal> && value){
+    s_logger->info("Estou no callback");
+    hyro::Signal analog_signal;
+    analog_signal = *value;
+    Thresholding digital_converter(analog_signal.value,2.0);
```

Comment:
Is the second argument amplitude? So, according to the exercise, this amplitude should be a argument.

Reply:
According to the exercise the SignalGeneratorComponent must have amplitude as a dynamic property, but there is nothing mentioning about the DigitalConverterComponent, by saying argument , do you mean dynamic property ?

Reply:
I solved this